lib/compilers/dex2oat.ts

```diff
@@ -211,14 +211,17 @@ export class Dex2OatCompiler extends BaseCompiler {
             'bootjars/apache-xml.jar',
         ];
 
+        const versionFile = this.artArtifactDir + '/snapshot-creation-build-number.txt';
+        const buildId = parseInt(fs.readFileSync(versionFile, {encoding: 'utf8'}));
+
+        const artVersion3310 = 9106705;
         const dex2oatOptions = [
             '--android-root=include',
             '--generate-debug-info',
             '--dex-location=/system/framework/classes.dex',
             `--dex-file=${d8DirPath}/${dexFile}`,
             '--copy-dex-files=always',
-            '--runtime-arg',
-            '-Xgc:CMC',
+            ...(buildId > artVersion3310 ? ['--force-allow-oj-inlines', '--runtime-arg', '-Xgc:CMC'] : []),
```

Comment:
Could you please check the version prefix (something like `prefix >= 34 || isLatest`) instead of checking the build ID? First, the build IDs are not incremental. Second, all versions starting with "33" don't support those flags.

Reply:
Done, thanks!